docs/spec/consensus/fork-accountability.md

```diff
@@ -0,0 +1,320 @@
+
+* If the underlying communication ensures that every message exchanged between correct full nodes is eventually received, this attack is easily detected at the main chain level
+
+* This is an attack on the full node level (Fork-Full).
+* It extends also to the lite clients,
+* For both we need a detection and recovery mechanism.
+
+#### Scenario 2: Equivocation to a lite client (LCS)
+
+
+Validators:
+* a set F of faulty validators with more than 2/3 of the voting power.
+
+Execution:
+* for the main chain F behaves nicely
+* F coordinates to sign a block B that is different from the one on the main chain.
+* the lite clients obtains B and trusts at as it is signed by more and 2/3 of the voting power.
+
+Consequences:
+
+Once equivocation is used to attack lite client it opens space
+for different kind of attacks as application state can be diverged in any direction. For example, it can modify validator set such that it contains only validators that do not have any stake bonded. Note that after a lite client is fooled by a fork, that means that an attacker can change application state and validator set arbitrarily.
+
+In order to detect such (equivocation-based attack), the lite client would need to cross check its state with some correct validator (or to obtain a hash of the state from the main chain using out of band channels).
+
+*Remark.* The lite client would be able to create evidence of misbehavior, but this would require to pull potentially a lot of data from correct full nodes. Maybe we need to figure out different architecture where a lite client that is attacked will push all its data for the current unbonding period to a correct node that will inspect this data and submit corresponding evidence. There are also architectures that assumes a special role (sometimes called fisherman) whose goal is to collect as much as possible useful data from the network, to do analysis and create evidence transactions. That functionality is outside the scope of this document.
+
+*Remark.* The difference between LCS and LCB might only be in the amount of voting power needed to convince lite client about arbitrary state. In case of LCB where security threshold is at minimum, an attacker can arbitrarily modify application state with  more than 1/3 of voting power, while in case of LCS it requires more than 2/3 of the voting power.
+
+### Flip-flopping: Amnesia based attacks
+
+
+
+In case of amnesia, faulty validators lock some value *v* in some round *r*, and then vote for different value *v'* in higher rounds without correctly unlocking value *v*. This attack can be used both on full nodes and lite clients.
+
+#### Scenario 3: At most 2/3 of faults
+
+Validators:
+
+* a set F of faulty validators with more than 1/3 but at most 2/3 of the voting power
+* a set C of correct validators
+
+Execution:
+
+* More than 2/3 of voting power (containing correct and faulty validators) commit a block A in round *r*.
+* All reach a round *r' > r*.
```

Comment:
Once a correct validator commit a block in some round, it does not proceed to higher rounds of that height. It starts new height with round 0.